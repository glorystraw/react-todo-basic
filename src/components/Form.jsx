import React from 'react';
import PropTypes from 'prop-types';
import CustomButton from './CustomButton';

class Form extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            title: ''
        };
        this.store = this.props.store;
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleSubmit(event) {
        event.preventDefault();
        let title = this.state.title;
        if (title) {
            this.props.onAdd(title);
            this.setState({ title: '' });
        }
    }

    handleChange(event) {
        let title = event.target.value;
        this.setState({ title });
    }

    render() {
        return (
            <form className="todo-form" onSubmit={this.handleSubmit}>
                <input
                    type="text"
                    value={this.state.title}
                    placeholder="Add new task" onChange={this.handleChange} />
                <CustomButton type="submit">Add</CustomButton>
            </form>
        );
    }
}

Form.propTypes = {
    onAdd: PropTypes.func.isRequired
};

export default Form;