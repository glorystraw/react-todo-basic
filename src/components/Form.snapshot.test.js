import React from 'react';
import renderer from 'react-test-renderer';
import Form from './Form';

test('Snapshot testing enabled Button component', () => {
  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});