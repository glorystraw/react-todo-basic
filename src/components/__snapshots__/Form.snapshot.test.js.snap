// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing enabled Button component 1`] = `
<form
  className="todo-form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Add new task"
    type="text"
    value=""
  />
  <button>
    Add
  </button>
</form>
`;
